src/logger/lwt_log_core.ml

```diff
@@ -47,6 +47,17 @@ let string_of_level = function
   | Error -> "error"
   | Fatal -> "fatal"
 
+let level_of_string str =
+  let str = (String.lowercase [@ocaml.warning "-3"]) str in
+  match str with
+  | "debug" -> Ok Debug
+  | "info" -> Ok Info
+  | "notice" -> Ok Notice
+  | "warning" -> Ok Warning
+  | "error" -> Ok Error
+  | "fatal" -> Ok Fatal
+  | _ -> Pervasives.Error (Printf.sprintf "invalid log level (%s)" str)
```

Comment:
I'd say for this function, to have it evaluate to a `level option`. The `Error` result doesn't carry more information than `None` – there is only one way to fail, and the caller already knows which string they passed in. But to address your question, yes, we should use `Result.result` instead of `Pervasives.result`, in general.